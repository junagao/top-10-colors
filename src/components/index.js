export { default as ColorList } from 'components/ColorList'
export { default as ColorItem } from 'components/ColorItem'
export { default as Header } from 'components/Header'
export { default as Rating } from 'components/Rating'
export { default as Star } from 'components/Star'
