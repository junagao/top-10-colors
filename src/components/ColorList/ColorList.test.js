import React from 'react'
import { shallow } from 'enzyme'
import { ColorList, ColorItem } from 'components'

describe('ColorList', () => {
  let wrapper

  const colorsMock = [
    {
      id: 'fc352593-c91d-4cf6-b689-7478ee2b61c8',
      name: 'orange',
      hex: '#FFA500',
      rgb: 'rgb(255, 165, 0)',
      rating: 3,
    },
    {
      id: 'b0e3fab2-0477-4e28-b2e0-c2281a21d09b',
      name: 'purple',
      hex: '#800080',
      rgb: 'RGB(128, 0, 128)',
      rating: 5,
    },
  ]

  beforeEach(() => {
    wrapper = shallow(<ColorList colors={colorsMock} onRate={jest.fn()} />)
  })

  it('should render correctly given the required props', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('should render one List element', () => {
    expect(wrapper.find('List').length).toEqual(1)
  })

  it('should render two li elements', () => {
    expect(wrapper.find('li').length).toEqual(2)
  })

  it('should render two instance of ColorItem components', () => {
    expect(wrapper.find(ColorItem).length).toEqual(2)
  })
})
