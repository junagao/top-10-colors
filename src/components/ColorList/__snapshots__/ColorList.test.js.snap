// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ColorList should render correctly given the required props 1`] = `
<List>
  <li
    key="fc352593-c91d-4cf6-b689-7478ee2b61c8"
  >
    <ColorItem
      hex="#FFA500"
      id="fc352593-c91d-4cf6-b689-7478ee2b61c8"
      name="orange"
      onRate={[MockFunction]}
      rating={3}
      rgb="rgb(255, 165, 0)"
    />
  </li>
  <li
    key="b0e3fab2-0477-4e28-b2e0-c2281a21d09b"
  >
    <ColorItem
      hex="#800080"
      id="b0e3fab2-0477-4e28-b2e0-c2281a21d09b"
      name="purple"
      onRate={[MockFunction]}
      rating={5}
      rgb="RGB(128, 0, 128)"
    />
  </li>
</List>
`;
